agent/ecscni/netconfig_linux.go

```diff
@@ -169,3 +171,46 @@ func NewAppMeshConfig(appMesh *appmesh.AppMesh, cfg *Config) (string, *libcni.Ne
 
 	return defaultAppMeshIfName, networkConfig, nil
 }
+
+// NewServiceConnectNetworkConfig creates a new ServiceConnect CNI network configuration
+func NewServiceConnectNetworkConfig(
+	scConfig *serviceconnect.Config,
+	enableIPv4, enableIPv6 bool,
+	cfg *Config) (string, *libcni.NetworkConfig, error) {
+	var ingressConfig []IngressConfigJSONEntry
+	for _, ic := range scConfig.IngressConfig {
+		newEntry := IngressConfigJSONEntry{
+			ListenerPort: ic.ListenerPort,
+		}
+		if ic.InterceptPort != nil {
+			newEntry.InterceptPort = *ic.InterceptPort
+		}
+		ingressConfig = append(ingressConfig, newEntry)
+	}
+
+	var egressConfig *EgressConfigJSON
+	if scConfig.EgressConfig != nil {
+		egressConfig = &EgressConfigJSON{
+			ListenerPort: scConfig.EgressConfig.ListenerPort,
+			VIP: VIPConfigJSON{
+				IPv4CIDR: scConfig.EgressConfig.VIP.IPV4CIDR,
+				IPv6CIDR: scConfig.EgressConfig.VIP.IPV6CIDR,
+			},
+		}
+	}
+
+	scNetworkConfig := ServiceConnectConfig{
```

Comment:
We also need to add `NetworkMode`

Reply:
Do you mean `RedirectMode`? Will do once the CNI plugin PR is merged. The CNI version contained in this PR does not have that `RedirectMode` yet.